### PR TITLE
[pickers] New `StaticDatePicker` component

### PR DESCRIPTION
**Pull request by flaviendelangle**
- [x] New `useStaticPicker` hook
- [x] New `StaticDatePicker` component using `useStaticPicker`
- [x] Remove the open / close related props from the `BasePickerProps` to avoid having them leaking on the static pickers.

#### Remove mask related props

- `acceptRegex`
- `disableMaskedInput`
- `ignoreInvalidInputs`
- `mask`
- `rifmFormatter`

#### Remove keyboard editing related props (no more keyboard mode on the mobile toolbar)

- `renderInput`
- `InputProps`
- `label`
- `inputFormat`
- `inputRef`

#### Remove props already not used in v5

- `InputAdornmentProps`
- `OpenPickerButtonProps`
- `closeOnSelect`
- `components.OpenPickerIcon`
- `getOpenDialogAriaText`
- `disableOpenPicker`

### CODE REVIEW

File: packages/x-date-pickers/src/internals/models/props/basePickerProps.tsx [hunk 1]
```diff
@@ -66,3 +66,7 @@ export interface BaseNextPickerProps<TValue, TDate, TView extends CalendarOrCloc
    */
   localeText?: PickersInputComponentLocaleText<TDate>;
 }
+
+/**
+ * Props common to all non-static pickers
+ */
```
Comment (alexfauquette) [inline packages/x-date-pickers/src/internals/models/props/basePickerProps.tsx:72]: Should it be moved on top of the `UsePickerValueNonStaticProps`?

**Review by alexfauquette — APPROVED** [review]
The code sounds good. I like the distinction between common, static, and non-static props

I did not test the component yet

File: packages/x-date-pickers/src/internals/models/props/basePickerProps.tsx (hunk 1 above)
Comment (flaviendelangle) [reply packages/x-date-pickers/src/internals/models/props/basePickerProps.tsx:72]: :laughing: Oh I forgot to remove this comment
I was going to create a `BaseNextNonStaticPickerProps` but in the end I went for the `MobileOnly` and `DesktopOnly` interfaces.
I'll remove this comment

Comment (flaviendelangle) [reply packages/x-date-pickers/src/internals/models/props/basePickerProps.tsx]: I can add a description there indeed

File: packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+   */
+  fixedWeekNumber: PropTypes.number,
+  /**
+   * Format of the date when rendered in the input(s).
+   */
+  inputFormat: PropTypes.string,
+  /**
+   * Pass a ref to the `input` element.
+   */
+  inputRef: PropTypes.oneOfType([
```
Comment (LukasTy) [inline packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx:125]: This one as well

File: packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    PropTypes.func,
+    PropTypes.shape({
+      current: PropTypes.object,
+    }),
+  ]),
+  /**
+   * The label content.
+   */
+  label: PropTypes.node,
```
Comment (LukasTy) [inline packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx:134]: And this one as well? 🤔

File: packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+  /**
+   * If `true`, today's date is rendering without highlighting with circle.
+   * @default false
+   */
+  disableHighlightToday: PropTypes.bool,
+  /**
+   * Do not render open picker button (renders only the field).
+   * @default false
+   */
+  disableOpenPicker: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx:101]: Good catch ! I'll do the same for `usePickerViews` to remove it

**Comment by flaviendelangle** [comment]
@LukasTy I removed all the props you found that do not belong on the static pickers.
It adds a little complexity, especially for props that are handled directly on the components like `inputRef` and `label` that I could maybe move to the headless wrappers to unify everything.

File: packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+  props: StaticNextDatePickerProps<TDate> & React.RefAttributes<HTMLDivElement>,
+) => JSX.Element) & { propTypes?: any };
+
+const VIEW_LOOKUP = {
+  day: renderDateView,
+  month: renderDateView,
+  year: renderDateView,
+};
+
+const StaticNextDatePicker = React.forwardRef(function StaticNextDatePicker<TDate>(
```
Comment (LukasTy) [inline packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx:20]: It does seem that we are no longer using the forwarded ref anywhere. 🤔 

Below is the diff resolving the following error:
```
Warning: forwardRef render functions accept exactly two parameters: props and ref. Did you forget to use the ref parameter?
```

```diff
diff --git a/packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx b/packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx
index a504802f4..f274989a8 100644
--- a/packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx
+++ b/packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx
@@ -1,4 +1,3 @@
-import * as React from 'react';
 import PropTypes from 'prop-types';
 import { StaticNextDatePickerProps } from './StaticNextDatePicker.types';
 import { useNextDatePickerDefaultizedProps } from '../NextDatePicker/shared';
@@ -7,17 +6,13 @@ import { useStaticPicker } from '../internals/hooks/useStaticPicker';
 import { CalendarPickerView } from '../internals/models';
 import { renderDateView } from '../internals/utils/viewRenderers';
 
-type StaticDatePickerComponent = (<TDate>(
-  props: StaticNextDatePickerProps<TDate> & React.RefAttributes<HTMLDivElement>,
-) => JSX.Element) & { propTypes?: any };
-
 const VIEW_LOOKUP = {
   day: renderDateView,
   month: renderDateView,
   year: renderDateView,
 };
 
-const StaticNextDatePicker = React.forwardRef(function StaticNextDatePicker<TDate>(
+const StaticNextDatePicker = function StaticNextDatePicker<TDate>(
   inProps: StaticNextDatePickerProps<TDate>,
 ) {
   const defaultizedProps = useNextDatePickerDefaultizedProps<
@@ -41,7 +36,7 @@ const StaticNextDatePicker = React.forwardRef(function StaticNextDatePicker<TDat
   });
 
   return renderPicker();
-}) as StaticDatePickerComponent;
+};
 
 StaticNextDatePicker.propTypes = {
   // ----------------------------- Warning --------------------------------

```

**Review by LukasTy — APPROVED** [review]
Great work 💪 
Just one more small fix. 🙈 

Hopefully we are not gonna revert back to having the option of toggling the edit mode between calendar and input/field as all the effort of segregating interfaces would go in vain 😆

File: packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx (hunk 5 above)
Comment (flaviendelangle) [reply packages/x-date-pickers/src/StaticNextDatePicker/StaticNextDatePicker.tsx:20]: I forgot to pass it to the root indeed :grimacing: , I'll do it

**Comment by flaviendelangle** [comment]
> Hopefully we are not gonna revert back to having the option of toggling the edit mode between calendar and input/field as all the effort of segregating interfaces would go in vain 

I did not understand :angel:

**Comment by flaviendelangle** [comment]
:laughing: oh indeed, except for some props like `disableOpenPicker` that were pure typing mistake en v5

**Pull request merged**